### PR TITLE
Removing experimental dependency from pipelines

### PR DESCRIPTION
**Pull request by shiftylogic**
- Moved BufferPool to Buffer primitives.
- Removed experimental assembly dependency from pipelines.

### CODE REVIEW

File: src/System.IO.Pipelines/System.IO.Pipelines.csproj [hunk 1]
```diff
@@ -9,7 +9,7 @@
   </PropertyGroup>
   <ItemGroup>
     <ProjectReference Include="..\System.Buffers.Primitives\System.Buffers.Primitives.csproj" />
-    <ProjectReference Include="..\System.Buffers.Experimental\System.Buffers.Experimental.csproj" />
+    <ProjectReference Include="..\System.Collections.Sequences\System.Collections.Sequences.csproj" />
```
Comment (ahsonkhan) [inline src/System.IO.Pipelines/System.IO.Pipelines.csproj]: What is the purpose of this change?

Comment (shiftylogic) [inline src/System.IO.Pipelines/System.IO.Pipelines.csproj]: Buffers.Experimental indirectly pulled in the Collections.Sequences project. Pipelines depended on both. We are breaking the dependency on Experimental, so I had to add an explicit dependency on Collections.Sequences.

**Comment by shiftylogic** [comment]
@davidfowl @KrzysztofCwalina

File: src/System.Buffers.Primitives/System/Buffers/OwnedPinnedArray.cs [hunk 2]
```diff
@@ -6,37 +6,6 @@
 
 namespace System.Buffers
 {
-    public class OwnedNativeMemory : OwnedMemory<byte>
```
Comment (davidfowl) [inline src/System.Buffers.Primitives/System/Buffers/OwnedPinnedArray.cs]: I actually think these implementations could stay in Experimental. /cc @KrzysztofCwalina ?

Comment (shiftylogic) [inline src/System.Buffers.Primitives/System/Buffers/OwnedPinnedArray.cs]: I misunderstood a comment you made when we were chatting. Thought you needed those types. I'll move them back.

**Comment by KrzysztofCwalina** [comment]
looks good.
